src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs

```diff
@@ -159,10 +166,34 @@ private Task<bool> NominateProjectAsync(string projectUniqueName, IVsProjectRest
             {
                 _logger.LogError(e.ToString());
                 TelemetryUtility.EmitException(nameof(VsSolutionRestoreService), nameof(NominateProjectAsync), e);
-                return Task.FromResult(false);
+                return false;
             }
         }
 
+        private async Task<ProjectNames> GetProjectNamesAsync(string projectUniqueName)
+        {
+            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
Should we use the [standard ThreadHelper](https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.shell.threadhelper?view=visualstudiosdk-2019) to make VS threading analyzers catch more errors?

```suggestion
            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Reply:
@dominoFire  When do we `NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync()` vs `ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync()` ?

Reply:
The analyzers are on the JoinableTaskFactory. 

In  fact `ThreadHelper.JoinableTaskFactory` should be actively avoided: https://github.com/microsoft/vs-threading/blob/master/doc/testing_vs.md#avoid-calling-threadhelperjoinabletaskfactory-directly

Reply:
I'd love to have the JTF as a class property or field, so it's more mockable and testable, but I don't know a good pattern for getting the JTF from a constructor. Marking as resolved given Nikolche's comment.

Reply:
The JTF is static :) 
That's what makes it testable. You can provide a custom JTF in tests if needed.